Applications/Spire/Include/Spire/Ui/TimeInputWidget.hpp

```diff
@@ -0,0 +1,83 @@
+#ifndef SPIRE_TIME_INPUT_WIDGET_HPP
+#define SPIRE_TIME_INPUT_WIDGET_HPP
+#include <QPainter>
+#include <QWidget>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Spire/Scalar.hpp"
+#include "Spire/Ui/Ui.hpp"
+
+namespace Spire {
+
+  //! Displays a time input widget.
+  class TimeInputWidget : public QWidget {
+    public:
+
+      //! Signals that the time has been modified.
+      using TimeChangeSignal = Signal<void (Scalar)>;
```

Comment:
We do not have a time input, we have either a time duration which represents an elapsed period of time, a datetime which represents a single point in time, or a timeofday which represents a time of day.

Furthermore none of these widgets should make use of a Scalar, it defeats the entire point of having the widget that transforms to/from Scalars.

Reply:
Will the DateInputWidget be separate from the DateTimeInputWidget?